java/create-instance/src/main/java/com/example/cloud/bigtable/createinstance/CreateInstanceMain.java

```diff
@@ -0,0 +1,195 @@
+import com.google.bigtable.admin.v2.Instance;
+import com.google.bigtable.admin.v2.Instance.Type;
+import com.google.bigtable.admin.v2.ListInstancesRequest;
+import com.google.bigtable.admin.v2.ListInstancesResponse;
+import com.google.cloud.bigtable.config.BigtableOptions;
+import com.google.cloud.bigtable.config.BigtableVersionInfo;
+import com.google.cloud.bigtable.config.CredentialOptions;
+import com.google.cloud.bigtable.config.Logger;
+import com.google.cloud.bigtable.config.RetryOptions;
+import com.google.cloud.bigtable.grpc.BigtableInstanceGrpcClient;
+import com.google.cloud.bigtable.grpc.io.ChannelPool;
+import com.google.cloud.bigtable.grpc.io.CredentialInterceptorCache;
+import io.grpc.ClientInterceptor;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import org.apache.hadoop.conf.Configuration;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import javax.net.ssl.SSLException;
+
+public class CreateInstanceMain{
+  /** Constant <code>LOG.</code> */
+  protected static final Logger LOG = new Logger(CreateInstanceMain.class);
+  
+  private static final int PORT = 443;
+  private static final String DEFAULT_HOST = "bigtableadmin.googleapis.com";
+  
+  
+  // 256 MB, server has 256 MB limit.
+  private static final int MAX_MESSAGE_SIZE = 1 << 28;
+  
+  private static BigtableOptions options;
+  private static BigtableInstanceGrpcClient instanceClient;
+  private static String projectId;
+  private static String instanceId;
+  private static String location;
+  private static String displayName;
+  private static String clusterName;
+  private static String instanceType;
+  private static String LOCATION_FORMAT;
+  private static String PARENT_FORMAT;
+  
+  public static void main(String[] args) throws Exception {
+    //Project Id under which instance will be created.
+    projectId = args[0];
+    LOG.info("projectId:" + projectId);
+    
+    //Instance id to be created.
+    instanceId = args[1];
+    LOG.info("instanceId:" + instanceId);
+    
+    //provide zone. store your data near the services that need it.
+    location = args[2];
+    
+    //display name for an instance.
+    displayName = args[3];
+    
+    //cluster name.
+    clusterName = args[4];
+    
+    //instance type.
+    instanceType = args[5];
+    
+    //zone format API understands. 
+    LOCATION_FORMAT = "projects/" + projectId + "/locations/" + location;
+    PARENT_FORMAT = "projects/" + projectId;
+    
+    CreateInstanceMain main = new CreateInstanceMain();
+    main.execute();
+  }
+  
+  private void execute() throws IOException {
+    init();
+    createInstance();
+    listInstances();
+  }
+  
+  private void init() throws IOException  {
+    options = BigtableOptionsFactoryCreateInstance
+        .fromConfiguration(new Configuration());
+    
+    //Create GRPC client.
+    instanceClient = 
```

Comment:
Please use a `BigtableSession` here instead, and then call `getInstanceAdminClient()`

Reply:
For creating BigtableSession it requires an InstanceId, that was the reason I avoided it. I thought for creating new instance, having an existing instance should not be required ?

Reply:
In BigtableSession.createChannelPool() method, there is this line of code creating prefix interceptor which requires instance name.

final ClientInterceptor prefixInterceptor =
        new GoogleCloudResourcePrefixInterceptor(options.getInstanceName().toString())

Reply:
I created a new version of the client.  Please use 1.3.1-SNAPSHOT and use BigtableSession. createInstanceClient(options)